aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageSubPathStrategyListTest.kt

```diff
@@ -0,0 +1,175 @@
+        lateinit var synchronousAuth: SynchronousAuth
+        private lateinit var first: File
+        private lateinit var second: File
+        private lateinit var third: File
+        private lateinit var fourth: File
+        private lateinit var fifth: File
+
+        /**
+         * Initialize mobile client and configure the storage.
+         * Upload the test files ahead of time.
+         */
+        @JvmStatic
+        @BeforeClass
+        fun setUpOnce() {
+            val context = ApplicationProvider.getApplicationContext<Context>()
+            WorkmanagerTestUtils.initializeWorkmanagerTestUtil(context)
+
+            synchronousAuth = SynchronousAuth.delegatingToCognito(context, AWSCognitoAuthPlugin())
+
+            // Get a handle to storage
+            storageCategory = TestStorageCategory.create(context, R.raw.amplifyconfiguration)
+            synchronousStorage = SynchronousStorage.delegatingTo(storageCategory)
+
+            // Upload test files
+            first = RandomTempFile(FIRST_FILE_NAME, SMALL_FILE_SIZE)
+            synchronousStorage.uploadFile(FIRST_FILE_PATH, first, StorageUploadFileOptions.defaultInstance())
+            second = RandomTempFile(SECOND_FILE_NAME, SMALL_FILE_SIZE)
+            synchronousStorage.uploadFile(SECOND_FILE_PATH, second, StorageUploadFileOptions.defaultInstance())
+            third = RandomTempFile(THIRD_FILE_NAME, SMALL_FILE_SIZE)
+            synchronousStorage.uploadFile(THIRD_FILE_PATH, third, StorageUploadFileOptions.defaultInstance())
+            fourth = RandomTempFile(FOURTH_FILE_NAME, SMALL_FILE_SIZE)
+            synchronousStorage.uploadFile(FOURTH_FILE_PATH, fourth, StorageUploadFileOptions.defaultInstance())
+            fifth = RandomTempFile(FIFTH_FILE_NAME, SMALL_FILE_SIZE)
+            synchronousStorage.uploadFile(FIFTH_FILE_PATH, fifth, StorageUploadFileOptions.defaultInstance())
+        }
+    }
+
+    @After
+    fun tearDown() {
+        synchronousStorage.remove("photos/2023/$FIRST_FILE_NAME", StorageRemoveOptions.defaultInstance())
+        synchronousStorage.remove("photos/2023/$SECOND_FILE_NAME", StorageRemoveOptions.defaultInstance())
+        synchronousStorage.remove("photos/2024/$THIRD_FILE_NAME", StorageRemoveOptions.defaultInstance())
+        synchronousStorage.remove("photos/2024/$FOURTH_FILE_NAME", StorageRemoveOptions.defaultInstance())
+        synchronousStorage.remove("photos/$FIFTH_FILE_NAME", StorageRemoveOptions.defaultInstance())
+    }
+
+    @Test
+    fun testListWithIncludeStrategyAndStoragePath() {
+        val path = StoragePath.fromString("public/photos/")
+        val options = AWSS3StoragePagedListOptions
+            .builder()
+            .setPageSize(10)
+            .setSubpathStrategy(SubpathStrategy.Include)
+            .build()
+
+        val result = synchronousStorage.list(path, options)
+
+        result.items.apply {
+            Assert.assertEquals(5, size)
+        }
```

Comment:
can we assert tha the `result.items` contains particular paths? It will help the reader see instead of looking back up at the paths defined above and guessing which of the 5 are included

Reply:
Good suggestion, I've updated the tests to check for paths and exclude paths.